pkg/scheduler/api/namespace_info.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2018 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+// NamespaceName is name of namespace
+type NamespaceName string
+
+const (
+	// NamespaceWeightKey is the key in ResourceQuota.spec.hard indicating the weight of this namespace
+	NamespaceWeightKey = "volcano.sh/namespace.weight"
+	// DefaultNamespaceWeight is the default weight of namespace
+	DefaultNamespaceWeight = 1
+)
+
+// NamespaceInfo records information of namespace
+type NamespaceInfo struct {
+	// Name is the name of this namespace
+	Name NamespaceName
+	// Weight is the highest weight among many ResourceQuota.
+	Weight int64
```

Comment:
if we use signed integer, what's the best practice here? Do we suggest users to always use weights > 0 or it could be negative numbers?

Reply:
Yes, weight < 1 is meaningless in this case, because we have a default value which is 1. Any weight < 1 will be ignored.